projects/RabbitMQ.Client/client/impl/SocketFrameHandler.cs

```diff
@@ -232,14 +212,21 @@ public void Close()
             }
         }
 
+        /*
         public InboundFrame ReadFrame()
         {
             return InboundFrame.ReadFrom(_reader, _frameHeaderBuffer);
         }
+        */
+
+        public ValueTask<InboundFrame> ReadFrame()
+        {
+            return InboundFrame.ReadFrom(_pipeReader, _frameHeaderBuffer);
+        }
 
         public void SendHeader()
         {
-            byte[] headerBytes = new byte[8];
+            byte[] headerBytes = ArrayPool<byte>.Shared.Rent(8);
```

Comment:
if you have a fixed small number of elements you should use stackalloc instead of rent its faster

Reply:
stackalloc creates a Span, and that doesn't work with Encoding.GetBytes on net461.